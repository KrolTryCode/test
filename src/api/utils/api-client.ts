import axios, { AxiosError, InternalAxiosRequestConfig } from 'axios';
import i18n from 'i18next';
import queryString from 'query-string';

import { ApiConfig, Api } from '~/api/utils/api-requests';
import { authPath, loginPath } from '~/utils/configuration/routes-paths';
import {
  projectLocalStorageService,
  projectSessionStorageService,
} from '~/utils/localstorage/project-storage/project-storage-instance';
import { ProjectStorageKey } from '~/utils/localstorage/project-storage/project-storage.types';
import { showErrorMessage } from '~/utils/show-error-message';

export interface ErrorResponse {
  code: number;
  message: string;
}

const getStorageService = () => {
  const rememberMe = projectLocalStorageService.get(ProjectStorageKey.RememberMe);
  return rememberMe ? projectLocalStorageService : projectSessionStorageService;
};

const BASE_API_URL = '/api';

const API_CONFIG: ApiConfig = {
  baseURL: BASE_API_URL,
};

const API_CONFIG_SECURE: ApiConfig = {
  baseURL: BASE_API_URL,
  securityWorker: async () => {
    const token = getStorageService().get(ProjectStorageKey.AccessToken);

    if (token) {
      return { headers: { Authorization: `Bearer ${token ?? ''}` } };
    }
  },
  paramsSerializer: params => {
    const keys = Object.keys(params);

    let res: {
      [key: string]: unknown;
    } = {};

    for (const key of keys) {
      if (typeof params[key] === 'object' && !Array.isArray(params[key])) {
        // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
        res = { ...res, ...params[key] };
      } else {
        // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
        res[key] = params[key];
      }
    }

    return queryString.stringify(res, { arrayFormat: 'none' });
  },
};

export const ApiClient = new Api(API_CONFIG);

export const ApiClientSecured = new Api(API_CONFIG_SECURE);

ApiClientSecured.instance.interceptors.request.use(request => {
  request.headers.set('Accept-Language', i18n.language);
  return request;
});

ApiClient.instance.interceptors.request.use(request => {
  request.headers.set('Accept-Language', i18n.language);
  return request;
});

// https://gist.github.com/Godofbrowser/bf118322301af3fc334437c683887c5f#file-axios-refresh_token-1-js
let isRefreshing = false;
let failedQueue: {
  resolve: (token: string | null) => void;
  reject: (error: AxiosError) => void;
}[] = [];

const processQueue = (error: AxiosError | null, token: string | null = null) => {
  failedQueue.forEach(prom => {
    error ? prom.reject(error) : prom.resolve(token);
  });
  failedQueue = [];
};

// Можно задать кастомные обработчики для каждой ошибки
ApiClientSecured.instance.interceptors.response.use(
  response => response,
  async (error: AxiosError) => {
    const originalRequest = error.config as InternalAxiosRequestConfig<any> & { _retry: boolean };

    if (error.response?.status === 502) {
      showErrorMessage(error, 'ERROR.TECHNICAL_ERROR');
    }
    if (error.response?.status === 401 && !originalRequest._retry) {
      if (error.config?.url === '/api/v1/jwt/refresh') {
        window.location = `/${authPath}/${loginPath}` as unknown as Location;
        return;
      }

      if (window.location.pathname.startsWith(`/${authPath}`)) {
        return Promise.reject(error);
      }

      if (isRefreshing) {
        return new Promise(function (resolve, reject) {
          failedQueue.push({ resolve, reject });
        })
          .then(token => {
            originalRequest.headers['Authorization'] = 'Bearer ' + String(token);
            return axios(originalRequest);
          })
          .catch(err => {
            return Promise.reject(err);
          });
      }
      originalRequest._retry = true;
      isRefreshing = true;

      const storage = getStorageService();
      storage.remove(ProjectStorageKey.AccessToken);
      const refreshToken = storage.get(ProjectStorageKey.RefreshToken);

      return new Promise(function (resolve, reject) {
        ApiClientSecured.tokenV1Controller
          .regenerateAccessToken({ headers: { Authorization: `Bearer ${refreshToken ?? ''}` } })
          .then(({ token }) => {
            storage.set(ProjectStorageKey.AccessToken, token!);
            axios.defaults.headers.common['Authorization'] = 'Bearer ' + token;
            processQueue(null, token);
            if (originalRequest) {
              originalRequest.headers['Authorization'] = 'Bearer ' + token;
              resolve(axios(originalRequest));
            }
          })
          .catch((err: AxiosError) => {
            processQueue(err, null);
            reject(err);
          })
          .finally(() => {
            isRefreshing = false;
          });
      });
    }
    return Promise.reject(error);
  },
);
