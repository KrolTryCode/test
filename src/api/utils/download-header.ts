import { RequestParams } from './api-requests';

export const getDownloadHeaderCSV = (): RequestParams => ({
  headers: {
    accept: 'text/csv',
  },
});
