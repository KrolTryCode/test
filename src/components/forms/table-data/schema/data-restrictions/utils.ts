import {
  ColumnValue,
  Const,
  FloatConst,
  IntConst,
  StringConst,
  StringLengthValue,
} from '~/api/utils/api-requests';

export const isColumnValueType = (value: any) => {
  return '@type' in value && value['@type'] === 'ColumnValue';
};

export const getRightOrLeftValue = (
  currentObject:
    | string
    | object
    | ColumnValue
    | Const
    | StringLengthValue
    | FloatConst
    | IntConst
    | StringConst,
) => {
  let value: object | string;
  if (typeof currentObject === 'object' && 'value' in currentObject) {
    value = currentObject.value;
  } else {
    value = currentObject;
  }
  return value;
};
