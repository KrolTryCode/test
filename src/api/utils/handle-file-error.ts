import { AxiosError } from 'axios';

import { showErrorMessage } from '~/utils/show-error-message';

export function handleFileUploadingError(e: unknown) {
  if (e instanceof AxiosError && e.response?.status === 413) {
    showErrorMessage(e, 'FILES.ERROR.TOO_BIG');
  } else {
    showErrorMessage(e, 'FILES.UPLOAD_FAILED');
  }
}
