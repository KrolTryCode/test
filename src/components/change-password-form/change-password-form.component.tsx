import { yupResolver } from '@hookform/resolvers/yup';
import { Stack } from '@mui/material';
import { Form, FormItem, FormButtons, Button, InputText } from '@pspod/ui-components';
import { FC } from 'react';
import { useForm } from 'react-hook-form';
import { useTranslation } from 'react-i18next';

import { useGetPasswordConfigurationQuery } from '~/api/queries/settings/get-password-configuration.query';
import { AccountConfiguration } from '~/api/utils/api-requests';
import { FormInputText } from '~/components/react-hook-form';

import {
  ChangePasswordForm as IChangePasswordForm,
  getSchema,
} from './change-password-form.schema';
import { RulesTooltip } from './rules-tooltip/rules-tooltip.component';

interface ChangePasswordFormProps {
  isPending?: boolean;
  username?: string;
  isSettingNew?: boolean;
  onReject?: () => void;
  onSave: (values: IChangePasswordForm) => void;
}

export const ChangePasswordForm: FC<ChangePasswordFormProps> = ({
  onSave,
  onReject,
  username = '',
  isPending = false,
  isSettingNew = false,
}) => {
  const { t } = useTranslation();

  const { data: passwordConfig } = useGetPasswordConfigurationQuery({
    placeholderData: {} as AccountConfiguration,
  });

  const {
    register,
    handleSubmit,
    control,
    formState: { isValid, isSubmitted, isLoading },
  } = useForm<IChangePasswordForm>({
    mode: 'onBlur',
    reValidateMode: 'onBlur',
    resolver: yupResolver(getSchema(passwordConfig!, t)),
  });

  return (
    <Form onSubmit={handleSubmit(onSave)} isRequired>
      <FormItem label={t('USER.EMAIL')} isHidden={!username}>
        <InputText value={username} isReadonly />
      </FormItem>
      <FormItem
        label={
          <Stack display={'inline-flex'} direction={'row'} alignItems={'center'} gap={'4px'}>
            <span>
              {isSettingNew
                ? t('COMMON.NEW', { type: t('AUTH.PASSWORD.NAME').toLowerCase() })
                : t('AUTH.PASSWORD.NAME')}
            </span>
            <RulesTooltip config={passwordConfig} />
          </Stack>
        }
      >
        <FormInputText
          type={'password'}
          autoComplete={'new-password'}
          controllerProps={{ ...register('password'), control }}
        />
      </FormItem>
      <FormItem label={t('AUTH.PASSWORD.REPEAT')}>
        <FormInputText
          type={'password'}
          autoComplete={'new-password'}
          controllerProps={{ ...register('confirmedPassword'), control }}
        />
      </FormItem>
      <FormButtons>
        <Button onClick={onReject} variant={'outlined'} color={'primary'}>
          {t('ACTION.CANCEL')}
        </Button>
        <Button
          type={'submit'}
          color={'primary'}
          variant={'contained'}
          disabled={!isValid && isSubmitted}
          isLoading={isPending || isLoading}
        >
          {t('AUTH.PASSWORD.CREATE')}
        </Button>
      </FormButtons>
    </Form>
  );
};
