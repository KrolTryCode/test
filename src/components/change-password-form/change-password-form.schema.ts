import { TFunction } from 'i18next';
import * as y from 'yup';

import { AccountConfiguration } from '~/api/utils/api-requests';
import { getPasswordSchema } from '~/utils/validation/schemas';

export interface ChangePasswordForm {
  password: string;
  confirmedPassword: string;
}

export const getSchema = (passwordConfig: AccountConfiguration, t: TFunction) =>
  y.object({
    password: getPasswordSchema(passwordConfig, t).required().default(''),
    confirmedPassword: y
      .string()
      .required()
      .when('password', ([password], schema) => {
        const pRe = new RegExp(`^${password}$`);
        return schema.matches(pRe, { message: { key: 'yup:custom.passwordMismatch' } });
      })
      .default(''),
  });
