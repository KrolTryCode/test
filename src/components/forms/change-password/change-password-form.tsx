import { yupResolver } from '@hookform/resolvers/yup';
import { Form, FormItem, FormButtons, Button, InputText } from '@pspod/ui-components';
import { useQuery } from '@tanstack/react-query';
import { FC } from 'react';
import { useForm } from 'react-hook-form';
import { useTranslation } from 'react-i18next';

import { getPasswordConfigurationQueryOptions } from '~/api/queries/settings/get-password-configuration.query';
import { AccountConfiguration } from '~/api/utils/api-requests';
import { LabelWithRules } from '~/components/label-with-rules/label-with-rules.component';
import { PasswordRuleList } from '~/components/password-rule-list/rule-list.component';
import { FormInputText } from '~/components/react-hook-form';

import {
  ChangePasswordForm as IChangePasswordForm,
  getSchema,
} from './change-password-form.schema';

interface ChangePasswordFormProps {
  isPending?: boolean;
  username?: string;
  isSettingNew?: boolean;
  onReject?: () => void;
  onSave: (values: IChangePasswordForm) => void;
}

export const ChangePasswordForm: FC<ChangePasswordFormProps> = ({
  onSave,
  onReject,
  username = '',
  isPending = false,
  isSettingNew = false,
}) => {
  const { t } = useTranslation();

  const { data: passwordConfig } = useQuery(getPasswordConfigurationQueryOptions());

  const {
    register,
    handleSubmit,
    control,
    formState: { isValid, isSubmitted, isLoading },
  } = useForm<IChangePasswordForm>({
    mode: 'onBlur',
    reValidateMode: 'onBlur',
    resolver: yupResolver(getSchema(passwordConfig ?? ({} as AccountConfiguration), t)),
  });

  return (
    <Form onSubmit={handleSubmit(onSave)} isRequired>
      <FormItem label={t('USER.EMAIL')} isHidden={!username}>
        <InputText value={username} isReadonly />
      </FormItem>
      <FormItem
        label={
          <LabelWithRules
            label={
              isSettingNew
                ? t('COMMON.NEW', { type: t('AUTH.PASSWORD.NAME').toLowerCase() })
                : t('AUTH.PASSWORD.NAME')
            }
            content={passwordConfig ? <PasswordRuleList config={passwordConfig} /> : undefined}
          />
        }
      >
        <FormInputText
          type={'password'}
          autoComplete={'new-password'}
          controllerProps={{ ...register('password'), control }}
        />
      </FormItem>
      <FormItem label={t('AUTH.PASSWORD.REPEAT')}>
        <FormInputText
          type={'password'}
          autoComplete={'new-password'}
          controllerProps={{ ...register('confirmedPassword'), control }}
        />
      </FormItem>
      <FormButtons>
        <Button onClick={onReject} variant={'outlined'} color={'primary'}>
          {t('ACTION.CANCEL')}
        </Button>
        <Button
          type={'submit'}
          color={'primary'}
          variant={'contained'}
          disabled={!isValid && isSubmitted}
          isLoading={isPending || isLoading}
        >
          {t('AUTH.PASSWORD.CREATE')}
        </Button>
      </FormButtons>
    </Form>
  );
};
