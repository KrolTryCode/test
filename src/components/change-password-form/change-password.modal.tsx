import { modal } from '@pspod/ui-components';
import { InstanceProps } from 'react-modal-promise';

import { UpdatePasswordRequest } from '~/api/utils/api-requests';

import { ChangePasswordForm } from './change-password-form.component';

export const changePasswordModal = ({
  title,
  user,
  onSave,
}: {
  title: string;
  user: string;
  onSave: (data: UpdatePasswordRequest) => void;
}) =>
  modal<UpdatePasswordRequest>({
    title,
    onOk: onSave,
    renderContent: (modalInstance: InstanceProps<UpdatePasswordRequest, never>) => (
      <ChangePasswordForm
        {...modalInstance}
        isSettingNew
        username={user}
        onSave={modalInstance.onResolve}
      />
    ),
  });
