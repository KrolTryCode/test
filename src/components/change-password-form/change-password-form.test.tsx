import { screen } from '@testing-library/react';
import { t } from 'i18next';
import { it, describe, expect } from 'vitest';

import { passwordConfiguration } from 'tests/__mocks__/api-client.mock';
import { renderComponent } from 'tests/utils/render-component';

import { ChangePasswordForm } from './change-password-form.component';
import {
  getSchema,
  ChangePasswordForm as IChangePasswordForm,
} from './change-password-form.schema';

const correctFormData: IChangePasswordForm = {
  password: 'abc!qA23',
  confirmedPassword: 'abc!qA23',
};

const onSave = vi.fn().mockReturnValue(() => Promise.resolve());

describe('Change password form', () => {
  let schema: ReturnType<typeof getSchema>;

  beforeAll(() => {
    renderComponent(<ChangePasswordForm username={'test@mail.com'} onSave={onSave} />, {
      withRouter: true,
    });

    schema = getSchema(passwordConfiguration, t);
  });

  it('should render the change password form', async () => {
    expect(await screen.findByText('Задать пароль')).toBeInTheDocument();
    const container = await screen.findByRole('form');
    expect(container).toMatchSnapshot();
  });

  it('should pass with correct data', async () => {
    expect(await schema.isValid(correctFormData)).toBeTruthy();
  });

  it('should fail with wrong data', async () => {
    expect(await schema.isValid({})).toBeFalsy();
    expect(await schema.isValid({ ...correctFormData, confirmedPassword: '123' })).toBeFalsy();

    const weakpas = 'weakpas';
    expect(
      await schema.isValid({ ...correctFormData, confirmedPassword: weakpas, password: weakpas }),
    ).toBeFalsy();
  });
});
