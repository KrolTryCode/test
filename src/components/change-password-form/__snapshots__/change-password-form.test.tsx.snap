// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Change password form > should render the change password form 1`] = `
<form
  class="MuiStack-root css-eir3lh-MuiStack-root"
  role="form"
>
  <div
    class="MuiFormControl-root css-1dixy4o-MuiFormControl-root"
  >
    <label
      class="MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-1xtmzjf-MuiFormLabel-root"
    >
      E-mail
      <span
        aria-hidden="true"
        class="MuiFormLabel-asterisk css-ylp2hm-MuiFormLabel-asterisk"
      >
         
        *
      </span>
    </label>
    <div
      class="MuiBox-root css-eal7pa"
    >
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-1w54es3-MuiFormControl-root-MuiTextField-root"
      >
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-rccg0t-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            aria-readonly="true"
            autocomplete="off"
            class="MuiInputBase-input MuiOutlinedInput-input css-1thzzby-MuiInputBase-input-MuiOutlinedInput-input"
            id=":r0:"
            readonly=""
            type="text"
            value="test@mail.com"
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-ihdtdm"
            >
              <span
                class="notranslate"
              >
                ​
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
  </div>
  <div
    class="MuiFormControl-root css-1dixy4o-MuiFormControl-root"
  >
    <label
      class="MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-1xtmzjf-MuiFormLabel-root"
    >
      <div
        class="MuiStack-root css-en0hi5-MuiStack-root"
      >
        <span>
          Пароль
        </span>
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
          data-mui-internal-clone-element="true"
          data-testid="HelpOutlineIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M11 18h2v-2h-2zm1-16C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2m0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8m0-14c-2.21 0-4 1.79-4 4h2c0-1.1.9-2 2-2s2 .9 2 2c0 2-3 1.75-3 5h2c0-2.25 3-2.5 3-5 0-2.21-1.79-4-4-4"
          />
        </svg>
      </div>
      <span
        aria-hidden="true"
        class="MuiFormLabel-asterisk css-ylp2hm-MuiFormLabel-asterisk"
      >
         
        *
      </span>
    </label>
    <div
      class="MuiBox-root css-eal7pa"
    >
      <div
        class="MuiBox-root css-0"
        data-mui-internal-clone-element="true"
      >
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-1w54es3-MuiFormControl-root-MuiTextField-root"
        >
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd css-1m46seq-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              aria-readonly="false"
              autocomplete="new-password"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-18al6m3-MuiInputBase-input-MuiOutlinedInput-input"
              id=":r3:"
              type="password"
              value=""
            />
            <div
              class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
            >
              <button
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-yvetkp-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeSmall css-1o004fu-MuiSvgIcon-root"
                  data-testid="VisibilityOutlinedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 6c3.79 0 7.17 2.13 8.82 5.5C19.17 14.87 15.79 17 12 17s-7.17-2.13-8.82-5.5C4.83 8.13 8.21 6 12 6m0-2C7 4 2.73 7.11 1 11.5 2.73 15.89 7 19 12 19s9.27-3.11 11-7.5C21.27 7.11 17 4 12 4m0 5c1.38 0 2.5 1.12 2.5 2.5S13.38 14 12 14s-2.5-1.12-2.5-2.5S10.62 9 12 9m0-2c-2.48 0-4.5 2.02-4.5 4.5S9.52 16 12 16s4.5-2.02 4.5-4.5S14.48 7 12 7"
                  />
                </svg>
              </button>
            </div>
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-ihdtdm"
              >
                <span
                  class="notranslate"
                >
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="MuiFormControl-root css-1dixy4o-MuiFormControl-root"
  >
    <label
      class="MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-1xtmzjf-MuiFormLabel-root"
    >
      Повтор пароля
      <span
        aria-hidden="true"
        class="MuiFormLabel-asterisk css-ylp2hm-MuiFormLabel-asterisk"
      >
         
        *
      </span>
    </label>
    <div
      class="MuiBox-root css-eal7pa"
    >
      <div
        class="MuiBox-root css-0"
        data-mui-internal-clone-element="true"
      >
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-1w54es3-MuiFormControl-root-MuiTextField-root"
        >
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd css-1m46seq-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              aria-readonly="false"
              autocomplete="new-password"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-18al6m3-MuiInputBase-input-MuiOutlinedInput-input"
              id=":r5:"
              type="password"
              value=""
            />
            <div
              class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
            >
              <button
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-yvetkp-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeSmall css-1o004fu-MuiSvgIcon-root"
                  data-testid="VisibilityOutlinedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 6c3.79 0 7.17 2.13 8.82 5.5C19.17 14.87 15.79 17 12 17s-7.17-2.13-8.82-5.5C4.83 8.13 8.21 6 12 6m0-2C7 4 2.73 7.11 1 11.5 2.73 15.89 7 19 12 19s9.27-3.11 11-7.5C21.27 7.11 17 4 12 4m0 5c1.38 0 2.5 1.12 2.5 2.5S13.38 14 12 14s-2.5-1.12-2.5-2.5S10.62 9 12 9m0-2c-2.48 0-4.5 2.02-4.5 4.5S9.52 16 12 16s4.5-2.02 4.5-4.5S14.48 7 12 7"
                  />
                </svg>
              </button>
            </div>
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-ihdtdm"
              >
                <span
                  class="notranslate"
                >
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="MuiStack-root css-1v1qdkn-MuiStack-root"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiLoadingButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorSecondary MuiButton-disableElevation MuiButton-root MuiLoadingButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorSecondary MuiButton-disableElevation  css-wg7wa7-MuiButtonBase-root-MuiButton-root-MuiLoadingButton-root"
      id=":r6:"
      tabindex="0"
      type="button"
    >
      Отменить
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiLoadingButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-disableElevation MuiButton-root MuiLoadingButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-disableElevation  css-83fcih-MuiButtonBase-root-MuiButton-root-MuiLoadingButton-root"
      id=":r7:"
      tabindex="0"
      type="submit"
    >
      Задать пароль
    </button>
  </div>
</form>
`;
